pandas/tests/frame/test_apply.py

```diff
@@ -785,6 +785,14 @@ def non_reducing_function(val):
             df.applymap(func)
             assert values == df.a.to_list()
 
+    def test_apply_with_byte_string(self):
+        # GH 34529
+        try:
+            df = pd.DataFrame(np.array([b"abcd", b"efgh"]), columns=["col"])
+            df.apply(lambda x: x.astype("object"))
```

Comment:
huh? don't use try/excepts in tests.

please also compare the constructed return object e.g. lke

result=
expected=
tm.assert_frame_equal(result, expected)

Reply:
I did the test with try/except because the problem is not if df.apply(lambda x: x.astype("object")) runs correctly but if it runs in the first place. Knowing that is assert_frame_equal still the way to go?

Reply:
yes

Reply:
we care about 'does it run', it it succeeds the test wont' immediately fail. we *also* and *always* care about the results, are they correct.

Reply:
ok, will be done!